processing/src/test/java/io/druid/segment/SchemalessIndex.java

```diff
@@ -438,33 +438,59 @@ private static QueryableIndex makeAppendedMMappedIndex(
         timeline.add(intervals.get(i), i, noneShardSpec.createChunk(filesToMap.get(i)));
       }
 
-      List<Pair<File, Interval>> intervalsToMerge = Lists.transform(
-          timeline.lookup(new Interval("1000-01-01/3000-01-01")),
-          new Function<TimelineObjectHolder<Integer, File>, Pair<File, Interval>>()
-          {
-            @Override
-            public Pair<File, Interval> apply(@Nullable TimelineObjectHolder<Integer, File> input)
-            {
-              return new Pair<File, Interval>(input.getObject().getChunk(0).getObject(), input.getInterval());
-            }
-          }
-      );
-
-      for (final Pair<File, Interval> pair : intervalsToMerge) {
-        adapters.add(
-            new RowboatFilteringIndexAdapter(
-                new QueryableIndexIndexableAdapter(IndexIO.loadIndex(pair.lhs)),
-                new Predicate<Rowboat>()
+      final List<IndexableAdapter> adapters = Lists.newArrayList(
+          Iterables.transform(
+              Iterables.concat(
+                  Iterables.transform(
```

Comment:
is it possible to break this up a little? the code is very hard to read when chaining tranform -> concat -> transform -> tansform